### PR TITLE
[JSDoc] Add missing documentation for botframework-streaming websocket and payloads folders

### PR DESCRIPTION
**Pull request by sw-joelmut**
Addresses # 2602

## Description
This PR introduces the change to `jsdoc/require-jsdoc` in the [.eslintrc.json](https://github.com/microsoft/botbuilder-js/blob/master/.eslintrc.json#L263) file from `warn` to `error` and adds all the missing JSDoc documentation for [botframework-streaming](https://github.com/microsoft/botbuilder-js/tree/master/libraries/botframework-streaming) based on the errors thrown. Some documentation is ported and adapted from [botbuilder-dotnet](https://github.com/microsoft/botbuilder-dotnet)

> **Note:** We divided the work for this library into two PRs to ease the review. PR# 150 contains the rest of the botbuilder fixes.

## Specific Changes

  - Private methods' documentation that does not exist on botbuilder-dotnet we added the `@private` tag in the JSDoc comment.
  - Added JSDoc comments in **15 files**.

## Testing
In the following image there is a sneak peek about the ESLint Report with no JSDoc errors.
image

### CODE REVIEW

**Review by iscaltritti-sw — APPROVED** [review]

File: libraries/botframework-streaming/src/payloads/payloadAssemblerManager.ts [hunk 1]
```diff
@@ -20,12 +20,23 @@ export class PayloadAssemblerManager {
...
+     */
     public constructor(streamManager: StreamManager, onReceiveResponse: Function, onReceiveRequest: Function) {
         this.streamManager = streamManager;
         this.onReceiveRequest = onReceiveRequest;
         this.onReceiveResponse = onReceiveResponse;
     }
 
+    /**
+     * Retrieves the assembler's payload as a stream.
+     * @param The Header of the Stream to retrieve.
```
Comment (ceciliaavila) [inline libraries/botframework-streaming/src/payloads/payloadAssemblerManager.ts:37]:
```suggestion
     * @param header The Header of the Stream to retrieve.

```

File: libraries/botframework-streaming/src/payloads/payloadAssemblerManager.ts [hunk 2]
```diff
@@ -20,12 +20,23 @@ export class PayloadAssemblerManager {
...
+     * @returns >A `SubscribableStream` of the assembler's payload.
```
Comment (ceciliaavila) [inline libraries/botframework-streaming/src/payloads/payloadAssemblerManager.ts:38]:
```suggestion
     * @returns A `SubscribableStream` of the assembler's payload.

```

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: libraries/botframework-streaming/src/payloads/payloadAssemblerManager.ts (hunk 2 above)
Comment (sw-joelmut) [reply libraries/botframework-streaming/src/payloads/payloadAssemblerManager.ts]: Resolved all feedback

**Review by ceciliaavila — APPROVED** [review]

File: libraries/botframework-streaming/src/payloads/payloadAssemblerManager.ts [hunk 3]
```diff
@@ -20,12 +20,25 @@ export class PayloadAssemblerManager {
...
         this.onReceiveRequest = onReceiveRequest;
         this.onReceiveResponse = onReceiveResponse;
     }
 
+    /**
+     * Retrieves the assembler's payload as a stream.
+     * @param header The Header of the Stream to retrieve.
+
+     * @returns A `SubscribableStream` of the assembler's payload.
+
```
Comment (santgr11) [inline libraries/botframework-streaming/src/payloads/payloadAssemblerManager.ts:40]: Fix these spaces

**Review by santgr11 — CHANGES_REQUESTED** [review]

**Comment by sw-joelmut** [comment]
Thanks @santgr11, feedback applied.

**Review by santgr11 — APPROVED** [review]

**Review by denscollo — APPROVED** [review]

**Comment by denscollo** [comment]
promoted

**Pull request closed without merging**